template/template_test.go

```diff
@@ -150,6 +150,55 @@ func Test_ExecuteComplexMessage(t *testing.T) {
 			},
 			expected: "Click here standalone ",
 		},
+		{
+			name: "matcher many",
```

Comment:
i would write new test function for mathcer, to reduce repetition, when we testing different match cases
something like that
```go
func Test_Matcher(t *testing.T) {
	t.Parallel()

	tests := []struct {
		name      string
		inputStr  string
		inputMaps []map[string]any
		expected  []string
		funcs     []registry.Func
	}{
		{
			name:     "matcher string",
			inputStr: `.match { $n :string } no {{no apples}} one {{{ $n } apple}} * {{{ $n } apples}}`,
			inputMaps: []map[string]any{
				{"n": "no"},
				{"n": "one"},
				{"n": "many"},
			},
			expected: []string{"no apples", "one apple", "many apples"},
		},
	}

	for _, tt := range tests {
		if len(tt.inputMaps) != len(tt.expected) {
			t.Error("Arguments and expected results should have the same length")
		}

		tt := tt
		t.Run(tt.name, func(t *testing.T) {
			t.Parallel()

			template, err := New().Parse(tt.inputStr)
			require.NoError(t, err)

			for i, inputMap := range tt.inputMaps {
				actual, err := template.Sprint(inputMap)
				require.NoError(t, err)

				require.Equal(t, tt.expected[i], actual)
			}
		})
	}
}
```